airflow/serialization/serialized_objects.py

```diff
@@ -162,9 +163,9 @@ def encode_timezone(var: Timezone) -> str | int:
     )
 
 
-def decode_timezone(var: str | int) -> Timezone:
+def decode_timezone(var: str | int) -> Timezone | FixedTimezone:
```

Comment:
FixedTimezone inherits from Timezone so we don’t actually need this union, just annotating the return value as `Timezone` would work. I wonder if this is guaranteed though (logically it should be…?)

Reply:
In pendulum 2 FixedTimezone subclass of  Timezone
In pendulum 3 both Timezone  and FixedTimezone  are subclasses of PendulumTimezone

So yeah, right now we could use Timezone annotation

Reply:
But not eventually huh. This is so bad. (Maybe we should bite the bullet and transition all the Airflow internals to rely on the standard `tzinfo` interface instead. That’s a big undertaking though and shouldn’t be in this PR. I guess this is the right annotation going forward.